src/components/nodes/messageFlow/messageFlow.vue

```diff
@@ -57,20 +56,6 @@ export default {
     targetIsPool() {
       return this.targetType === poolId;
     },
-    targetIsInDifferentPool() {
-      if (this.targetIsPool()) {
-        return true;
-      }
-
-      if (this.targetIsIntermediateCatchEvent()) {
-        return true;
-      }
-
-      const targetPool = this.target.component.node.pool;
-      const sourcePool = this.sourceShape.component.node.pool;
-
-      return sourcePool != null && sourcePool !== targetPool;
-    },
```

Comment:
I don't think these validation rules should be removed. See this comment regarding connection rules: https://github.com/ProcessMaker/modeler/issues/220#issuecomment-462840889.

Looking at the issue you linked to, you should only be adding the following new behaviour (emphasis mine):
> connect a message flow **from an empty pool** to a **message catch event** or **other task types** inside a process inside another pool

Although removing this now allows the connection specified above, it also now allows previously invalid connection. Check out page 42 (page 72 using the page input): https://www.omg.org/spec/BPMN/2.0/PDF/. It shows a list of valid  connection rules for message flows.